frontend/src/metabase/query_builder/components/filters/FilterPopover.jsx

```diff
@@ -55,6 +55,20 @@ export default class FilterPopover extends Component<*, Props, State> {
         tableMetadata: PropTypes.object.isRequired
     };
 
+    componentWillMount() {
+        window.addEventListener('keydown', this.commitOnEnter.bind(this));
```

Comment:
Need to remove `.bind(this)` otherwise you're a passing different versions of `commitOnEnter` to `removeEventListener` and it won't be removed.